tests/Kiota.Builder.Tests/Extensions/OpenApiUrlTreeNodeExtensionsTests.cs

```diff
@@ -673,4 +675,336 @@ public void GetsClassNameWithSegmentsToSkipForClassNames()
+                            Schema = new OpenApiSchema {
+                                Type = "string"
+                            }
+                        },
+                    },
+                    Operations = {
+                        [OperationType.Get] = new OpenApiOperation
+                        {
+                            Responses = new OpenApiResponses {
+                                ["200"] = new OpenApiResponse
+                                {
+                                    Content = {
+                                        ["application/json"] = new OpenApiMediaType
+                                        {
+                                            Schema = userSchema
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+                ["users/{user-id}/manager"] = new OpenApiPathItem
+                {
+                    Operations = {
+                        [OperationType.Get] = new OpenApiOperation
+                        {
+                            Parameters = {
+                                new OpenApiParameter {
+                                    Name = "user-id",
+                                    In = ParameterLocation.Path,
+                                    Required = true,
+                                    Schema = new OpenApiSchema {
+                                        Type = "string"
+                                    }
+                                },
+                            },
+                            Responses = new OpenApiResponses {
+                                ["200"] = new OpenApiResponse
+                                {
+                                    Content = {
+                                        ["application/json"] = new OpenApiMediaType
+                                        {
+                                            Schema = userSchema
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+            },
+            Components = new OpenApiComponents
+            {
+                Schemas = new Dictionary<string, OpenApiSchema> {
+                    {
+                        "microsoft.graph.user", userSchema
+                    }
+                }
+            }
+        };
+        var mockLogger = new CountLogger<KiotaBuilder>();
+        var builder = new KiotaBuilder(mockLogger, new GenerationConfiguration { ClientClassName = "Graph", ApiRootUrl = "https://localhost" }, _httpClient);
+        var node = builder.CreateUriSpace(document);
+        node.MergeIndexNodesAtSameLevel(mockLogger);
+        var usersCollectionIndexNode = GetChildNodeByPath(node, "users/{foo-id}");
```

Comment:
`{foo-id}` doesn't sound right to me, in this test, we are picking up `foo` instead of `id` or `user-id`

Reply:
that's because:
- we pick things up alphabetically, which is because we're not running "words proximity" between the path parameter candidate names and the parent segment name
- we're appending -id when not present (on a single path parameter for the segment)

We could stop appending -id if you think that makes more sense?

Reply:
> we pick things up alphabetically

This is not correct, the problem is caused by the fact that we are deciding to deduplicate the nodes and we make the arbitrary decision to rename them as the (alphabetically sorted) first entry.